draft-ietf-oauth-selective-disclosure-jwt.md

```diff
@@ -274,70 +291,176 @@ The array is created as follows:
+does not depend on the original order of elements.
 
-It is the Issuer who decides which claims are selectively disclosable and which are not. However, claims controlling the validity of the SD-JWT, such as `iss`, `exp`, or `nbf` are usually included in plaintext. End-User claims MAY be included as plaintext as well, e.g., if hiding the particular claims from the Verifier does not make sense in the intended use case.
+For example, using the digest of the object property Disclosure created above,
+the Issuer could create the following SD-JWT payload to make `family_name`
+selectively disclosable:
 
-Claims that are not selectively disclosable are included in the SD-JWT in plaintext just as they would be in any other JWT.
+```json
+{
+  "given_name": "Alice",
+  "_sd": ["uutlBuYeMDyjLLTpf6Jxi7yNkEF35jdyWMn9U7b_RYY"]
+}
+```
 
-Selectively disclosable claims are omitted from the SD-JWT. Instead, the digests of the respective Disclosures and potentially decoy digests are contained as an array in a new JWT claim, `_sd`.
+### Array Elements
 
-The `_sd` key MUST refer to an array of strings, each string being a digest of a Disclosure or a decoy digest as described above.
+Digests of Disclosures for array elements are added to the array in the same
+position as the original claim value in the array. For each digest, an object
+of the form `{"...": "<digest>"}` is added to the array. The key MUST be the
+string `...` (three dots). The value MUST be the digest of the Disclosure as
+described in (#hashing_disclosures). There MUST NOT be any other keys in the
+object.
 
-The array MAY be empty in case the Issuer decided not to selectively disclose any of the claims at that level. However, it is RECOMMENDED to omit the `_sd` key in this case to save space.
+For example, using the digest of the array element Disclosure created above,
+the Issuer could create the following SD-JWT payload to make the second element
+of the `nationalities` array selectively disclosable:
 
-The Issuer MUST hide the original order of the claims in the array. To ensure this, it is RECOMMENDED to shuffle the array of hashes, e.g., by sorting it alphanumerically or randomly. The precise method does not matter as long as it does not depend on the original order of elements.
+```json
+{
+  "nationalities":
+    ["DE", {"...": "w0I8EKcdCtUPkGCNUrfwVp2xEgNjtoIDlOxc9-PlOhs"}]
+}
+```
 
-Issuers MUST NOT issue SD-JWTs where
+As described in (#verifier_verification), Verifiers ignore all selectively
+disclosable array elements for which they did not receive a Disclosure. In the
+example above, the verification process would output an array with only one
+element unless a matching Disclosure for the second element is received.
 
- * the key `_sd` is already used for the purpose other than to contain the array of digests, or
- * the same Disclosure value appears more than once (in the same array or in different arrays).
+## Example 1: SD-JWT {#example-1}
```

Comment:
I don't disagree this section probably belongs here in terms of the order, but the level of this section feels off (looked at [the rendered ver of this branch](https://drafts.oauth.net/oauth-selective-disclosure-jwt/danielfett/arrays/draft-ietf-oauth-selective-disclosure-jwt.html#name-format-of-an-sd-jwt)).

I think there is a logical separations between sections from 5.1 to 5.5 (basic operations) and from sections 5.6 to 5.7 (advanced concepts) from 5.8 to 5.9 (content of the SD-JWT) and 5.10 to 5.11 (combined formats). and having them all at the same level makes it hard to read..

Reply:
Yes, I agree! There will be other changes coming that may prompt a slight restructuring of existing sections. Let's revisit after those changes. I created Issue #290 for this.